flex/validation/common.py

```diff
@@ -323,13 +324,46 @@ def validate_allof_anyof(value, sub_schemas, context, method, **kwargs):
 
 
 def generate_allof_validator(allOf, context, **kwargs):
-    return functools.partial(validate_allof_anyof, sub_schemas=allOf, context=context, method=all)
+    if 'resolved_refs' not in context:
+        context['resolved_refs'] = []
+
+    unresolved_refs = []
+    for ref in allOf:
+        if isinstance(ref, dict) and '$ref' in ref:
+            schema = ref['$ref']
+            if schema not in context['resolved_refs']:
+                unresolved_refs.append(ref)
+                context['resolved_refs'].append(schema)
+        else:
+            unresolved_refs.append(ref)
+
+    return functools.partial(validate_allof_anyof, sub_schemas=unresolved_refs,
+                             context=context, method=all)
 
 
 def generate_anyof_validator(anyOf, context, **kwargs):
     return functools.partial(validate_allof_anyof, sub_schemas=anyOf, context=context, method=any)
 
 
+def add_polymorphism_requirements(object, schema, context, schema_validators):
+    object_type = None
```

Comment:
I think this line can be removed as the following `try/except` guarantees that `object_type` is either set or an exception is raised.